tests/unit/cluster/cli.tcl

```diff
@@ -81,6 +81,28 @@ start_multiple_servers 3 [list overrides $base_conf] {
 
     set node1_rd [redis_deferring_client 0]
 
+    test "use previous hostip in \"cluster-preferred-endpoint-type unknown-endpoint\" mode" {
+        
+        # backup and set cluster-preferred-endpoint-type unknown-endpoint
+        $node1_rd CONFIG GET cluster-preferred-endpoint-type
+        set endpoint_type_before_set [lindex [split [$node1_rd read] " "] 1]
+        $node1_rd CONFIG SET cluster-preferred-endpoint-type unknown-endpoint
+        $node1_rd read
+
+        # when redis-cli not in cluster mode, return MOVE with empty host
+        set slot_for_foo [exec src/redis-cli -c -p [srv 0 port] CLUSTER KEYSLOT foo]
```

Comment:
We already have a client connection in $node1. We can use it instead of exec redis-cli for all things except testing that redis-cli follows the redirect.

```suggestion
        set slot_for_foo [$node1 CLUSTER KEYSLOT foo]
```